integration/testrunner/grpc.go

```diff
@@ -230,10 +274,19 @@ func testGRPCEndpoint(ctx context.Context, t *testing.T, g *GRPC) {
 		} else {
 			t.Errorf("Did not get err as expected. Instead got reply: %v", reply)
 		}
-
-		query := g.makeQueryForGRPCStream(id)
+		wantNum := 4
+		query := g.makeQueryForGRPC(id)
+		t.Log(query.Q)
 		// we expect to have 4 audit logs - the last sent number (5) will be after the err occurred.
-		queryIfAuditLogsExistWithRetries(ctx, t, query, g.Config, "stream_fail_on_four", int64(4))
+		results := queryIfAuditLogsExistsWithRetries(ctx, t, query, g.Config, "stream_fail_on_four")
+		if len(results) != wantNum {
+			t.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", totalNumbers, len(results))
+		} else {
```

Comment:
Do we need such else here? I thought we don't need `else` here

Reply:
Yes, if `len(results) != wantNum` and we keep diffing each out, there will be panic for accessing array out of bounds problems.

Reply:
If len(results) != wantNum, should keep comparing the logentry, we can just return

Reply:
My idea here is that even the number doesn't match, it would still be good to check if fields matches, which would provide more context for debug when tests fail.

Reply:
Fixed.